src/main/java/eu/de4a/connector/mock/controller/DOController.java

```diff
@@ -1,33 +1,64 @@
 package eu.de4a.connector.mock.controller;
 
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
 import com.helger.commons.error.level.EErrorLevel;
+import eu.de4a.connector.mock.Helper;
+import eu.de4a.connector.mock.config.DOConfig;
 import eu.de4a.connector.mock.exampledata.CanonicalEvidenceExamples;
 import eu.de4a.connector.mock.exampledata.DataOwner;
 import eu.de4a.connector.mock.exampledata.EvidenceID;
+import eu.de4a.connector.mock.preview.PreviewMessage;
+import eu.de4a.connector.mock.preview.PreviewStorage;
 import eu.de4a.iem.jaxb.common.types.*;
 import eu.de4a.iem.xml.de4a.DE4AMarshaller;
 import eu.de4a.iem.xml.de4a.DE4AResponseDocumentHelper;
 import eu.de4a.iem.xml.de4a.IDE4ACanonicalEvidenceType;
 import eu.de4a.kafkaclient.DE4AKafkaClient;
 import lombok.extern.slf4j.Slf4j;
+import org.apache.commons.io.IOUtils;
+import org.apache.http.HttpResponse;
+import org.apache.http.client.fluent.Request;
+import org.apache.http.entity.ContentType;
+import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.context.annotation.Profile;
 import org.springframework.http.HttpStatus;
 import org.springframework.http.MediaType;
 import org.springframework.http.ResponseEntity;
+import org.springframework.messaging.simp.SimpMessagingTemplate;
+import org.springframework.scheduling.TaskScheduler;
 import org.springframework.web.bind.annotation.PostMapping;
 import org.springframework.web.bind.annotation.RequestMapping;
 import org.springframework.web.bind.annotation.RestController;
 import org.w3c.dom.Element;
 
+import java.beans.SimpleBeanInfo;
+import java.io.IOException;
 import java.io.InputStream;
+import java.io.StringWriter;
+import java.nio.charset.StandardCharsets;
+import java.time.Instant;
 import java.util.UUID;
+import java.util.concurrent.CompletableFuture;
+import java.util.function.Consumer;
 
 @RestController
 @Slf4j
 @RequestMapping(consumes = MediaType.APPLICATION_XML_VALUE, produces = MediaType.APPLICATION_XML_VALUE)
 @Profile("do")
 public class DOController {
 
+    @Autowired
+    private DOConfig doConfig;
+    @Autowired
+    private TaskScheduler taskScheduler;
+    @Autowired
+    private PreviewStorage previewStorage;
+    @Autowired
+    private SimpMessagingTemplate websocketMessaging;
+    @Autowired
+    private ObjectMapper objectMapper;
+
```

Comment:
Maybe add som DE4AKafkaCLient.send() messages so there is the possibility to follow message flows? And errors if any

Reply:
my bad. missed them...